sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -0,0 +1,172 @@
+//==------- kernel_bundle_impl.hpp - SYCL kernel_bundle_impl ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/backend_types.hpp>
+#include <CL/sycl/context.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/device.hpp>
+#include <CL/sycl/kernel_bundle.hpp>
+#include <detail/device_image_impl.hpp>
+#include <detail/program_manager/program_manager.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstring>
+#include <memory>
+#include <vector>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+class kernel_bundle_impl {
+
+public:
+  kernel_bundle_impl(const context &Ctx, const std::vector<device> &Devs,
+                     bundle_state State)
+      : MContext(Ctx), MDevices(Devs) {
+
+    MDeviceImages = detail::ProgramManager::getInstance().getSYCLDeviceImages(
+        Ctx, Devs, State);
+  }
+
+  kernel_bundle_impl(const context &Ctx, const std::vector<device> &Devs,
+                     const std::vector<kernel_id> &KernelIDs,
+                     bundle_state State)
+      : kernel_bundle_impl(Ctx, Devs, State) {
+
+    // Filter out images that have no kernel_ids specified
+    auto It = std::remove_if(MDeviceImages.begin(), MDeviceImages.end(),
+                             [&KernelIDs](const device_image_plain &Image) {
+                               const auto It = std::find_if(
```

Comment:
Nit: I think `std::any_of` is better suited here

Reply:
Replaced with none_of.